src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution_ArgsToParameters.cs

```diff
@@ -469,5 +477,35 @@ private static bool CanBeOptional(ParameterSymbol parameter, bool isMethodGroupC
 
             return null;
         }
+
+        private static int? CheckForDuplicateNamedArguments(AnalyzedArguments arguments)
+        {
+            if (arguments.Names.IsEmpty())
+            {
+                // No checks if there are no named arguments
+                return null;
+            }
+
+            var alreadyDefined = PooledHashSet<string>.GetInstance();
```

Comment:
>GetInstance(); [](start = 55, length = 14)

Should we pass in `arguments.Names.Count` as the initial capacity here?

Reply:
ObjectPool does not consider initial capacity :( Maybe we can revise that design later.

---
In reply to: [178680788](https://github.com/dotnet/roslyn/pull/25730#discussion_r178680788) [](ancestors = 178680788)